zzk/service/hoststate_test.go

```diff
@@ -1455,6 +1456,211 @@ func (t *ZZKTest) TestHostStateListener_Shutdown_Spawn(c *C) {
 	c.Assert(err, IsNil)
 }
 
+// Test Case: Start a container, restart it
+func (t *ZZKTest) TestHostStateListener_Spawn_StartRestart(c *C) {
+	conn := setUpServiceAndHostPaths(c)
+	handler := &mocks.HostStateHandler{}
+	shutdown := make(chan interface{})
+
+	req := StateRequest{
+		HostID:     hostId,
+		ServiceID:  serviceId,
+		InstanceID: 1,
+	}
+	err := CreateState(conn, req)
+	c.Assert(err, IsNil)
+
+	cancel := make(chan interface{})
+	listener := NewHostStateListener(handler, hostId, shutdown)
+	listener.SetConnection(conn)
+
+	// Start a container
+	ssdat := &ServiceState{
+		ContainerID: containerId,
+		ImageUUID:   imageId,
+		Paused:      false,
+		Started:     time.Now(),
+	}
+	var retShutdown <-chan interface{} = shutdown
+	containerExit := make(chan time.Time, 1)
+	var retExit <-chan time.Time = containerExit
+
+	handler.On("AttachContainer", mock.AnythingOfType("*service.ServiceState"), serviceId, 1).Return(nil, nil)
+	handler.On("StartContainer", retShutdown, serviceId, 1).Return(ssdat, retExit, nil)
```

Comment:
Can we put a .Once() on this so it will fail if StartContainer is called later during the restart?